homeassistant/components/nest/climate_sdm.py

```diff
@@ -279,7 +278,7 @@ def _get_supported_features(self) -> int:
             features |= ClimateEntityFeature.TARGET_TEMPERATURE
         if ThermostatEcoTrait.NAME in self._device.traits:
             features |= ClimateEntityFeature.PRESET_MODE
-        if FanTrait.NAME in self._device.traits:
+        if FanTrait.NAME in self._device.traits and self.hvac_mode != HVACMode.OFF:
```

Comment:
Supported features should not change during entity lifetime. This condition looks like it could cause the features to change.

Reply:
Yes that's right. Fan mode only works with specific HVAC nodes.

Should let the user try and just throw home assistant errors instead?

Reply:
I've updated this to leave supported features, but then have the error check when turning on the fan mode:

<img width="653" alt="Fan Mode" src="https://user-images.githubusercontent.com/6026418/173379402-9e6ca7ff-8a63-4cee-92bb-f06d3cdbcfd5.png">